### PR TITLE
Add Serverless token authorizer middleware

### PR DESCRIPTION
**Pull request by vutoan1245**
Rest API endpoints will now require a valid token.

For frontend local dev, use: http://localhost:5000/dev/mock_launch to fake a lti launch. It has 1 as the default userId and courseId

### CODE REVIEW

File: backend/src/handlers/auth/authorizer.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+): Promise<any> => {
+	if (!event.authorizationToken) {
+		return callback(null, {
+			principalId: 'user',
+			policyDocument: getPolicyDocument('Deny', event.methodArn),
+		});
+	}
+
+	try {
+		const token = event.authorizationToken.split(' ')[1];
```
Comment (kennygperez) [inline backend/src/handlers/auth/authorizer.ts:23]: This is the same logic thats in the oauth util.
try DRY

**Review by kennygperez — CHANGES_REQUESTED** [review]
Make sure to reuse code

**Review by Goff-Davis — COMMENTED** [review]
LGTM, see if you can do or dismiss kenny's request and I'll approve

**Review by kennygperez — DISMISSED** [review]

**Review by Goff-Davis — CHANGES_REQUESTED** [review]
Can you add that link you commented to the frontend readme?

**Review by Goff-Davis — APPROVED** [review]

**Review by kennygperez — APPROVED** [review]

**Pull request merged**